java/vector/src/main/java/org/apache/arrow/vector/ipc/message/ArrowRecordBatch.java

```diff
@@ -79,29 +79,13 @@ public ArrowRecordBatch(int length, List<ArrowFieldNode> nodes, List<ArrowBuf> b
       arrowBuffers.add(new ArrowBuffer(offset, size));
       LOGGER.debug("Buffer in RecordBatch at {}, length: {}", offset, size);
       offset += size;
-      if (alignBuffers && offset % 8 != 0) { // align on 8 byte boundaries
-        offset += 8 - (offset % 8);
+      if (alignBuffers) { // align on 8 byte boundaries
+        offset = (offset + 7) & 0xfffffffffffffff8L;
       }
     }
     this.buffersLayout = Collections.unmodifiableList(arrowBuffers);
   }
 
-  // clone constructor
-  private ArrowRecordBatch(boolean dummy, int length, List<ArrowFieldNode> nodes, List<ArrowBuf> buffers) {
-    this.length = length;
-    this.nodes = nodes;
-    this.buffers = buffers;
-    this.closed = false;
-    List<ArrowBuffer> arrowBuffers = new ArrayList<>();
-    long offset = 0;
-    for (ArrowBuf arrowBuf : buffers) {
-      long size = arrowBuf.readableBytes();
```

Comment:
it looks like this doesn't call   arrowBuf.getReferenceManager().retain();

That seems like a functionality difference?

Reply:
Also this changes removing this causes the buffers to be aligned by default, where they where unaligned before.

Reply:
Nice catch.

This constructor is functionally different from the public constructors. That is why it is in the code. 
I have restored this constructor and added comments for it.